File: tests/fixtures/scala/sample.scala
package fixture

import scala.collection.mutable

/** 形状基类（Scala 用 trait + class 表达）。 */
trait Shape {
  def area: Double
}

class Circle(val radius: Double) extends Shape {
  override def area: Double = if (radius > 0) math.Pi * radius * radius else 0.0
}

object Registry {
  private val items = mutable.ArrayBuffer.empty[Shape]

  def register(s: Shape): Unit = {
    items += s
  }
}

case class Point(x: Int, y: Int)

enum Kind {
  case Red, Green
}

object Helper {
  def helper(a: Int): Int = {
    if (a > 0) a else -a
  }
}

File: tests/fixtures/swift/sample.swift
import Foundation

/// 形状基类（Swift 用协议 + 类表达）。
protocol Shape {
    func area() -> Double
}

/// 圆形。
class Circle: Shape {
    private let radius: Double

    init(radius: Double) {
        self.radius = radius
    }

    func area() -> Double {
        return radius > 0 ? Double.pi * radius * radius : 0
    }
}

struct Point {
    var x: Int
    var y: Int
}

enum Kind {
    case red
    case green
}

func helper(_ a: Int) -> Int {
    if a > 0 && a < 10 {
        return a
    }
    return -a
}
